pkg/render/applicationlayer/applicationlayer.go

```diff
@@ -0,0 +1,368 @@
+// limitations under the License.
+
+package applicationlayer
+
+import (
+	_ "embed"
+	"fmt"
+	"strconv"
+	"strings"
+
+	ocsv1 "github.com/openshift/api/security/v1"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/components"
+	"github.com/tigera/operator/pkg/ptr"
+	"github.com/tigera/operator/pkg/render"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	"github.com/tigera/operator/pkg/render/common/secret"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	L7LogCollectorDeamonsetName = "l7-log-collector"
+	L7CollectorContainerName    = "l7-collector"
+	ProxyContainerName          = "envoy-proxy"
+	EnvoyLogsVolumeName         = "envoy-logs"
+	EnvoyConfigMapName          = "envoy-config"
+	EnvoyConfigMapKey           = "envoy-config.yaml"
+	APLName                     = "application-layer"
+)
+
+func ApplicationLayer(
+	pullSecrets []*corev1.Secret,
+	installation *operatorv1.InstallationSpec,
+	osType rmeta.OSType,
+	logsEnabled bool,
+	logIntervalSeconds *int64,
+	logRequestsPerInterval *int64,
+) render.Component {
+	return &component{
+		pullSecrets:            pullSecrets,
+		installation:           installation,
+		osType:                 osType,
+		logsEnabled:            logsEnabled,
+		logIntervalSeconds:     logIntervalSeconds,
+		logRequestsPerInterval: logRequestsPerInterval,
+	}
+}
+
+type component struct {
+	pullSecrets            []*corev1.Secret
+	installation           *operatorv1.InstallationSpec
+	osType                 rmeta.OSType
+	proxyImage             string
+	collectorImage         string
+	envoyConfigMap         *corev1.ConfigMap
+	logsEnabled            bool
+	logRequestsPerInterval *int64
+	logIntervalSeconds     *int64
+}
+
+func (c *component) ResolveImages(is *operatorv1.ImageSet) error {
+	reg := c.installation.Registry
+	path := c.installation.ImagePath
+	prefix := c.installation.ImagePrefix
+
+	if c.osType != c.SupportedOSType() {
+		return fmt.Errorf("l7 log collection is supported only on %s", c.SupportedOSType())
+	}
+
+	var err error
+	var errMsgs []string
+
+	c.proxyImage, err = components.GetReference(components.ComponentEnvoyProxy, reg, path, prefix, is)
+
+	if err != nil {
+		errMsgs = append(errMsgs, err.Error())
+	}
+	c.collectorImage, err = components.GetReference(components.ComponentL7Collector, reg, path, prefix, is)
+
+	if err != nil {
+		errMsgs = append(errMsgs, err.Error())
+	}
+
+	if len(errMsgs) != 0 {
+		return fmt.Errorf(strings.Join(errMsgs, ","))
+	}
+
+	return nil
+}
+
+func (c *component) SupportedOSType() rmeta.OSType {
+	return rmeta.OSTypeLinux
+}
+
+func (c *component) Objects() ([]client.Object, []client.Object) {
+	var objs []client.Object
+	// TODO: ensure that namespace exists before proceeding
```

Comment:
I don't think this TODO is necessary if we have the controller verifying that variant is set to TigeraSecureEnterprise in the controller.